src/ossl/aes.rs

```diff
@@ -224,26 +217,18 @@ impl AesOperation {
                     }
                 }
                 Ok(AesParams {
-                    iv: unsafe {
-                        std::slice::from_raw_parts(
-                            (*gcm_params).pIv,
-                            (*gcm_params).ulIvLen as usize,
-                        )
-                        .to_vec()
-                    },
+                    iv: bytes_to_vec!(unsafe { (*gcm_params).pIv }, unsafe {
+                        (*gcm_params).ulIvLen
+                    }),
                     maxblocks: 0,
                     ctsmode: 0,
                     datalen: 0,
-                    aad: unsafe {
-                        if (*gcm_params).ulAADLen > 0 {
-                            std::slice::from_raw_parts(
-                                (*gcm_params).pAAD,
-                                (*gcm_params).ulAADLen as usize,
-                            )
-                            .to_vec()
-                        } else {
-                            Vec::new()
-                        }
+                    aad: if unsafe { (*gcm_params).ulAADLen } > 0 {
+                        bytes_to_vec!(unsafe { (*gcm_params).pAAD }, unsafe {
+                            (*gcm_params).ulAADLen
```

Comment:
just a nit: Using temporary variable  for aadlen, similarly as done in the other cases, would make this block slightly more readable.

Reply:
Good point, fixed and updated